src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml

```diff
@@ -49,49 +55,57 @@
         <div class="namespace-view">
             @if (Model.CodeFile != null)
             {
-            <partial name="Shared/Navigation" model=" Model.CodeFile.Navigation" />
+                <partial name="Shared/Navigation" model=" Model.CodeFile.Navigation" />
             }
         </div>
     </div>
 
     <div class="col-10">
         <table class="code-window border rounded-1">
             <tbody>
-                @foreach (var line in Model.Lines) {
+                @foreach (var line in Model.Lines)
+                {
                 <tr class="code-line" data-line-id="@line.ElementId">
                     <td class="line-comment-button-cell">
-                        @if (line.ElementId != null) {
-                        <a class="line-comment-button">+</a>
-                        }
-                    </td>
-                    <td class="code"><span class="code-inner">@Html.Raw(line.DisplayString)</span></td>
-                </tr>
-                @if (line.Diagnostics.Any())
-                {
-                <tr class="code-diagnostics" data-line-id="@line.ElementId">
-                    <td colspan="2">
-                        @foreach (var lineDiagnostic in line.Diagnostics)
+                        @if (line.ElementId != null)
                         {
-                        <p>
-                            @if (lineDiagnostic.Text.StartsWith("DO"))
-                            {
-                            @:✅ <b>DO</b> @lineDiagnostic.Text.Substring(2)
-                            }
-                            else
+                            <a class="line-comment-button">+</a>
+                            @if (Model.Comments.TryGetThreadForLine(line.ElementId, out var _) || line.Diagnostics.Any())
                             {
-                            @lineDiagnostic.Text
+                                <span class="hidden icon icon-comments">💬</span>

```

Comment:
I wonder if we can keep this entirely client-side.

Reply:
I considered this but figured it was more performant to set on the server when we know there is a comment. I've added comments in the client side code that mentions where the icon is added in the server code.